File: frontend/src/types/api_types/taskSchemas.ts
import { WorkflowDefinition } from './workflowSchemas'

export type TaskStatus = 'PENDING' | 'RUNNING' | 'COMPLETED' | 'FAILED' | 'CANCELED'

export interface TaskResponse {
    id: string
    run_id: string
    node_id: string
    parent_task_id?: string
    status: TaskStatus
    inputs?: any
    outputs?: any
    error?: string
    start_time?: string
    end_time?: string
    subworkflow?: WorkflowDefinition
    subworkflow_output?: Record<string, any>
}

File: backend/app/schemas/task_schemas.py
from typing import Dict, Optional, Any
from pydantic import BaseModel
from datetime import datetime
from ..models.task_model import TaskStatus
from .workflow_schemas import WorkflowDefinitionSchema


class TaskResponseSchema(BaseModel):
    id: str
    run_id: str
    node_id: str
    parent_task_id: Optional[str]
    status: TaskStatus
    inputs: Optional[Any]
    outputs: Optional[Any]
    error: Optional[str]
    start_time: Optional[datetime]
    end_time: Optional[datetime]
    subworkflow: Optional[WorkflowDefinitionSchema]
    subworkflow_output: Optional[Dict[str, Any]]

    class Config:
        from_attributes = True  # Enable ORM mode
